router_engine.go

```diff
@@ -18,7 +21,7 @@ func NewEngine(cfg config.ServiceConfig, logger logging.Logger, w io.Writer) *gi
 	}
 
 	engine := gin.New()
-	engine.Use(gin.LoggerWithConfig(gin.LoggerConfig{Output: w}), gin.Recovery())
+	engine.Use(zapLogger(), gin.Recovery())
```

Comment:
Ideally I think this would leverage the `logger` passed into this function but that would probably be worse right now since the `Logger` interface is really basic.

Reply:
yeah, i agree on both points.